MdePkg/Include/Base.h

```diff
@@ -885,8 +885,14 @@ STATIC_ASSERT (ALIGNOF (__VERIFY_UINT32_ENUM_SIZE) == sizeof (__VERIFY_UINT32_EN
 **/
 #define BASE_CR(Record, TYPE, Field)  ((TYPE *) ((CHAR8 *) (Record) - OFFSET_OF (TYPE, Field)))
 
-// FIXME: Upstream general variants of these macros.
-#define DEBUG_RAISE()     ASSERT (FALSE)
+#define DEBUG_RAISE()                                                                                   \
+    if ((PcdGet8 (PcdDebugRaisePropertyMask) & DEBUG_PROPERTY_DEBUG_ASSERT_ENABLED) != 0) {             \
+      ASSERT (FALSE);                                                                                   \
+    } else if ((PcdGet8 (PcdDebugRaisePropertyMask) & DEBUG_PROPERTY_DEBUG_PRINT_ENABLED) != 0) {       \

```

Comment:
These shall not be `else if` clauses. You first do prints, then you assert, then you dead loop.